core/src/main/java/org/sonarsource/sonarlint/core/smartnotifications/SmartNotifications.java

```diff
@@ -0,0 +1,120 @@
+
+  public SmartNotifications(ConfigurationRepository configurationRepository, ConnectionConfigurationRepository connectionRepository,
+    ServerApiProvider serverApiProvider, SonarLintClient client) {
+    this.configurationRepository = configurationRepository;
+    this.connectionRepository = connectionRepository;
+    this.serverApiProvider = serverApiProvider;
+    this.client = client;
+  }
+
+  public void initialize(Path storageRoot) {
+    this.lastEventPollingService = new LastEventPolling(storageRoot);
+    smartNotificationsPolling = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "Smart Notifications Polling"));
+    smartNotificationsPolling.scheduleAtFixedRate(this::poll, 1, 60, TimeUnit.SECONDS);
+  }
+
+  private void poll() {
+    var notifications = new ArrayList<ServerNotification>();
+    var filteredBindings = configurationRepository.getBoundBindingPerConfigScopeId();
+
+    var projectKeysByConnectionId = filteredBindings.values().stream()
+      .collect(Collectors.groupingBy(BindingConfiguration::getConnectionId));
+
+    for (var projectKeyByConnectionId : projectKeysByConnectionId.entrySet()) {
+      var connectionId = projectKeyByConnectionId.getKey();
+      var connection = connectionRepository.getEndpointParams(connectionId);
+      var httpClient = serverApiProvider.getServerApi(connectionId);
+
+      if (connection.isPresent() && httpClient.isPresent()) {
+        var serverApi = httpClient.get();
+        var developersApi = serverApi.developers();
+
+        if (developersApi.isSupported()) {
+          var projectKeysByLastEventPolling =
+            projectKeyByConnectionId.getValue().stream()
+              .map(BindingConfiguration::getSonarProjectKey)
+              .distinct()
+              .collect(Collectors.toMap(Function.identity(), p -> lastEventPollingService.getLastEventPolling(p, connectionId)));
+
+          notifications.addAll(developersApi.getEvents(projectKeysByLastEventPolling)
+            .stream().map(e -> new ServerNotification(
+              e.getCategory(),
+              e.getMessage(),
+              e.getLink(),
+              e.getProjectKey(),
+              e.getTime()))
+            .collect(Collectors.toList()));
+        }
+      }
+    }
+
+    var scopeIds = new HashMap<String, List<String>>();
+    for (var bindingPerScopeId : filteredBindings.entrySet()) {
+      scopeIds.getOrDefault(bindingPerScopeId.getValue().getSonarProjectKey(), new ArrayList<>()).add(bindingPerScopeId.getKey());
+    }
+
+    for (var n : notifications) {
+      var smartNotification = new ShowSmartNotificationParams(SmartNotificationType.INFORMATION,
+        n.message(), n.link(), scopeIds.get(n.projectKey()));
+      client.showSmartNotification(smartNotification);
+    }
+  }
+
+  public void shutdown() {
+    smartNotificationsPolling.shutdownNow();
+  }
+
+  private static ZonedDateTime getLastNotificationTime(ZonedDateTime lastTime) {
```

Comment:
I don't think this is used, should it be ?

Reply:
Yep it should be used, (based on current implementation) we shouldn't retrieve more than a day old of notifications in case the last polling event is older than one day. I forgot to add this :smile: